indigo_app/js/components/toc-controller/TOCItem.vue

```diff
@@ -1,159 +0,0 @@
-              type="button"
-      >
-        <slot name="collapse-icon" v-if="expanded">
-          <span class="minus">-</span>
-        </slot>
-        <slot name="expand-icon" v-else>
-          <span class="plus">+</span>
-        </slot>
-      </button>
-    </div>
-    <div class="toc-item__content">
-      <div :class="`toc-item__content__action ${item.selected ? 'active' : ''}`">
-        <button @click="$emit('on-title-click', item.index)"
-                type="button"
-                class="toc-item__content__action__btn"
-        >
-          {{item.title}}
-        </button>
-        <div class="right-icon">
-          <slot name="right-icon" :item="item"></slot>
-        </div>
-      </div>
-      <ol v-if="isParent" v-show="expanded" class="" ref="children-container">
-        <TOCItem
-            v-for="(childItem, index) in item.children"
-            :item="childItem"
-            :key="index"
-            :ref="`child_item_${index}`"
-            :items-render-from-filter="itemsRenderFromFilter"
-            @on-title-click="(emittedIndex) => $emit('on-title-click', emittedIndex)"
-        >
-          <template v-slot:right-icon="props">
-            <slot name="right-icon" :item="props.item"></slot>
-          </template>
-          <template v-slot:expand-icon><slot name="expand-icon"></slot></template>
-          <template v-slot:collapse-icon><slot name="collapse-icon"></slot></template>
-        </TOCItem>
-      </ol>
-    </div>
-  </li>
-</template>
-
-<script>
-export default {
-  name: 'TOCItem',
-  data: () => ({
-    expanded: true
-  }),
-  props: {
-    item: {
-      type: Object,
-      default: () => ({})
-    },
-    itemsRenderFromFilter: {
-      type: Array,
-      default: () => []
-    },
-    index: {
-      type: Number,
-      default: 0
-    }
-  },
-
-  computed: {
-    isParent () {
-      return this.item.children && this.item.children.length;
-    },
-    showItem () {
-      // Show everything because search field is empty
-      if (!this.itemsRenderFromFilter.length) {
-        return true;
-      }
-      return this.itemsRenderFromFilter.length && this.itemsRenderFromFilter.some(item => item.title === this.item.title);
-    }
-  },
-
-  methods: {
-    toggleExpandedState () {
-      if (this.isParent) this.expanded = !this.expanded;
-    },
-    expandItemAndDescendants () {
-      this.expanded = true;
-      if (this.item.children && this.item.children.length) {
-        this.item.children.forEach((_, index) => {
-          this.$refs[`child_item_${index}`][0].expandItemAndDescendants();
-        });
-      }
-    },
-
-    collapseItemAndDescendants () {
-      this.expanded = false;
-      if (this.item.children && this.item.children.length) {
-        this.item.children.forEach((_, index) => {
-          this.$refs[`child_item_${index}`][0].collapseItemAndDescendants();
-        });
-      }
-    }
-  }
-};
-</script>
-
-<style scoped>
-.toc-item {
```

Comment:
Check what you need to preserve of all of this styling, so we don't lose it.

Reply:
done: 
note though that I put margin-left: 5px on `content__action__append` instead of the `content__action__title` element. the reason being is that `.content__action__append` conditionally renders so I would rather assign a margin gap to the element that dissapear rather that having a permanent margin-right on `content__action__title`